cpp/src/binaryop/compiled/operation.cuh

```diff
@@ -431,6 +431,21 @@ struct NullEquals {
   __device__ inline auto operator()(TypeLhs x, TypeRhs y) -> decltype(x == y);
 };
 
+struct NullNotEquals {
+  template <typename TypeLhs, typename TypeRhs>
+  __device__ inline auto operator()(
+    TypeLhs x, TypeRhs y, bool lhs_valid, bool rhs_valid, bool& output_valid) -> decltype(x != y)
+  {
+    output_valid = true;
+    if (!lhs_valid && !rhs_valid) return false;
+    if (lhs_valid && rhs_valid) return x != y;
+    return true;
```

Comment:
First: can't NullNotEquals be implemented as `!NullEquals()` somehow?  If not, I still think this can be simpler (and more legible):

```suggestion
    if (lhs_valid && rhs_valid) return x != y;
    return lhs_valid || rhs_valid;
```

Likewise for `NullEquals::operator()`. 

```
    output_valid = true;
    if (lhs_valid && rhs_valid) return x == y;
    return !lhs_valid && !rhs_valid;
```

Reply:
Good idea, done both.